ETTrace/Tracer/EMGTracer.mm

```diff
@@ -181,42 +181,50 @@ + (void)recordStackForAllThreads
         }
     }
 
+    std::map<thread_t, Stack *> stackMap;
+
     CFTimeInterval time = CACurrentMediaTime();
     for (mach_msg_type_number_t i = 0; i < thread_count; i++) {
         if (threads[i] == sETTraceThread) {
             continue;
         }
 
-        Stack stack;
-        stack.time = time;
-        FIRCLSWriteThreadStack(threads[i], stack.frames, kMaxFramesPerStack, &(stack.frameCount));
+        Stack *stack = new Stack;
+        stack->time = time;
+        FIRCLSWriteThreadStack(threads[i], stack->frames, kMaxFramesPerStack, &(stack->frameCount));
+        stackMap.insert(std::pair<unsigned int, Stack *>(threads[i], stack));
+    }
+
+    for (mach_msg_type_number_t i = 0; i < thread_count; i++) {
+        if (threads[i] != sETTraceThread)
+            thread_resume(threads[i]);
+    }
 
-        std::vector<Stack> *threadStack;
-        sThreadsLock.lock();
-        if (sThreadsMap->find(threads[i]) == sThreadsMap->end()) {
-            Thread *thread = [self createThread:threads[i]];
+    std::vector<Stack> *threadStack;
+    std::map<thread_t, Stack *>::iterator it;
+    sThreadsLock.lock();
+    for (it = stackMap.begin(); it != stackMap.end(); it++) {
```

Comment:
Now that the threads are resumed, copy the stacks into sThreadMap within the lock